interface/ceed-basis.c

```diff
@@ -621,6 +625,75 @@ int CeedBasisCreateH1(Ceed ceed, CeedElemTopology topo, CeedInt num_comp,
   return CEED_ERROR_SUCCESS;
 }
 
+/**
+  @brief Create a non tensor-product basis for H(div) discretizations
+  @param ceed        A Ceed object where the CeedBasis will be created
+  @param topo        Topology of element, e.g. hypercube, simplex, ect
+  @param num_comp    Number of componenet, we have 1 vector componenet in H(div)
+  @param num_nodes   Total number of nodes (dofs per element)
+  @param num_qpts    Total number of quadrature points
+  @param basis_space 2 for H(div) discretization (1 for H^1, 3 for H(curl))
+  @param interp      Row-major (dim*num_qpts * num_nodes) matrix expressing the values of
+                       nodal basis functions at quadrature points
+  @param div        Row-major (num_qpts * num_nodes) matrix expressing
+                       divergence of nodal basis functions at quadrature points
+  @param q_ref       Array of length num_qpts holding the locations of quadrature
+                       points on the reference element [-1, 1]
```

Comment:
The reference element will generally be something else.

Reply:
Yeah, I just copied from H^1 discretizations (line 568), perhaps we need to change that.

Reply:
Yeah, it was probably copied from the tensor constructor.

Reply:
I deleted [-1,1] on non-tensor basis constructor.